intermediate_source/torch_export_tutorial.py

```diff
@@ -365,6 +426,29 @@ def forward(self, x, y):
 except Exception:
     tb.print_exc()
 
+######################################################################
+# We can also describe one dimension in terms of other.
+
+class DerivedDimExample(torch.nn.Module):
```

Comment:
There's some crazier stuff that Avik's built into derived dims that users probably don't expect to work, but actually do work, maybe it's worth putting those on display? Stuff like (from test_export.py)

```
class Foo(torch.nn.Module):
    def forward(self, z, y):
        return z[1:] + y[1::3]

foo = Foo()

z, y = torch.randn(4), torch.randn(10)

dx = torch.export.Dim("dx", min=3, max=6)
dz = dx + 1
dy = dz * 3 + 1  # works
ep = export(
    foo,
    (z, y),
    dynamic_shapes=({0: dimz}, {0: dimy}),
)
assert ep.module()(torch.randn(7), torch.randn(19)).size()[0] == 6
```

We also allow static dim specs (e.g. dimx = 5, crashes if the value is wrong), as just a nice to have, maybe that's worth adding in a previous basic shapes example? I can also help with that.